### PR TITLE
feat: add run-many

### PR DESCRIPTION
**Pull request by mbriggs**
- want some feedback on this, commented out affected.spec.ts, that still needs to be implemented
 - _think_ it would be a good idea to make `--projects` and `--all` to be exclusive/required
 - add a new command (run-many) that basically is the same as affected, but allows you to specify projects to run
 - re-structure affected.ts so that the bulk of the functionality can be shared

### CODE REVIEW

File: packages/workspace/src/command-line/nx-commands.ts [hunk 1]
```diff
@@ -286,6 +293,54 @@ function withAffectedOptions(yargs: yargs.Argv): yargs.Argv {
...
+    })
+    .options('only-failed', {
+      describe: 'Isolate projects which previously failed',
+      type: 'boolean',
+      default: false
+    })
+    .option('verbose', {
+      describe: 'Print additional error stack trace on failure'
+    })
+    .option('plain', {
```
Comment (vsavkin) [inline packages/workspace/src/command-line/nx-commands.ts]: I don't think `affected:apps` and `affected:libs` will work with run-many. And this flag is only used by the two commands.

**Review by vsavkin — APPROVED** [review]
It looks good. The only two things are reenabling unit tests and adding some e2e tests.

**Comment by mbriggs** [comment]
- added --all, made it mutually exclusive with --projects, but required
 - yargs has super weird array input, changed --projects to be comma delimited
 - show warnings when asking to run a project that doesn't have configuration for target
 - show errors when asking to run a project that doesn't exist in the project
 - change the default reporter copy to not say "affected" (think whats there works fine in both cases)
 - rebase on to latest master (includes some bug fixes to affected.ts)

Still need to add tests

File: packages/workspace/src/command-line/run-tasks/run-command.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    createTask({
+      project,
+      target: nxArgs.target,
+      configuration: nxArgs.configuration,
+      overrides: overrides
+    })
+  );
+
+  const tasksMap: TasksMap = {};
+  projects.forEach(project => {
```
Comment (vsavkin) [inline packages/workspace/src/command-line/run-tasks/run-command.ts]: TaskMap is a set of all possible tasks, so it cannot be derived from the list of projects you are currently running.

Comment (vsavkin) [reply packages/workspace/src/command-line/run-tasks/run-command.ts]: In the current version is done like this:

Object.entries(affectedMetadata.dependencyGraph.projects)

So we generate a task map from the list of all possible projects

File: packages/workspace/src/command-line/run-tasks/run-command.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+import * as yargs from 'yargs';
+
+interface TasksMap {
+  [projectName: string]: { [targetName: string]: Task };
+}
+
+type RunArgs = yargs.Arguments & ReporterArgs;
+
+export function runCommand<T extends RunArgs>(
+  projects: ProjectNode[],
```
Comment (vsavkin) [inline packages/workspace/src/command-line/run-tasks/run-command.ts]: I'd rename it to indicate that this isn't list of all possible projects but only the projects are running tasks against this time.

File: packages/workspace/src/command-line/run-tasks/run-many.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  return args;
+}
+
+function projectListToGraph(projects: ProjectNode[]): DependencyGraph {
+  const roots = projects.map(p => p.name);
+  const map: ProjectMap = {};
+  projects.forEach(p => (map[p.name] = p));
+
+  return {
+    projects: map,
```
Comment (vsavkin) [inline packages/workspace/src/command-line/run-tasks/run-many.ts]: DependencyGraph must contain all projects. This map is only the projects you are running tasks against.

File: packages/workspace/src/command-line/run-tasks/run-many.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  if (!args.projects) {
+    args.projects = [];
+  } else {
+    args.projects = args.projects.split(',');
+  }
+
+  return args;
+}
+
+function projectListToGraph(projects: ProjectNode[]): DependencyGraph {
```
Comment (vsavkin) [inline packages/workspace/src/command-line/run-tasks/run-many.ts]: DependencyGraph describes the workspace. It must contain all projects, all dependenciesd and real roots. This function doesn't do this.

File: packages/workspace/src/command-line/run-tasks/run-command.ts [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+  projectGraph: DependencyGraph,
```
Comment (vsavkin) [inline packages/workspace/src/command-line/run-tasks/run-command.ts]: why not call it dependencyGraph?

**Review by vsavkin — CHANGES_REQUESTED** [review]
Left some feedback. I think there is an issue with the dep-graph in run-many. The rest looks good.

**Pull request merged**